Global sidebar - add tooltip-bottom-left style.

Addison-Stavlo opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to # https://github.com/Automattic/dotcom-forge/issues/5690 

## Proposed Changes

* The global style sidebar specific style rules for `.tooltip-bottom` cause the tooltips for search and notifs to render outside of the sidebar area. This updates this style definition in the global sidebar from .tooltip-bottom to .tooltip-bottom-left to signify left alignment, and is used with the WordPress.com menu item. Search and notif tooltips still default to .tooltip-bottom, but no longer get the left alignment overrides.
* This does not remove the `.tooltip-bottom` class as suggested in the original issue, as the component for notifications here is set up well to handle adding these classes based on prop values. The class name is still semantic and accurate for the tooltip here, and I think we should continue to add it - the main problem was the specific styles we set for it in the global sidebar.

BEFORE
![tooltips-old](https://github.com/Automattic/wp-calypso/assets/28742426/bf7e1eab-a5af-466d-b1b1-948b4c6d0e38)


AFTER
![tooltips-good-2](https://github.com/Automattic/wp-calypso/assets/28742426/7687ba9a-6aa8-421d-9d5b-11fc4b4225ee)



## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Build this calypso branch.
* Go to the /sites page and test the tooltip placements in the sidebar.
* Select an atomic site with classic view and test the tooltip placement in global site view.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Addison-Stavlo committed:
remove tooltip-bottom style overrides

Addison-Stavlo commented:
> We need to keep the CSS definition for .tooltip-bottom. What we need to remove is the CSS class on these two DOM elements.

The CSS definition that is removed is specifically within the global sidebar. I assume this was copied over from masterbar tooltips or other areas.

These sites and notifs are ‘reusable’ components that take a tooltip and tooltipPosition properties, and default to ‘bottom’ when none is specifically added. They are tooltips that are bottom positioned, so the class name seems semantic as well. Given all that, i don't think it makes sense to remove the .tooltip-bottom class from these.

davemart-in commented:
The problem if we remove it is that the tooltip on the logo is no longer aligned properly:

<img width="186" alt="CleanShot 2024-02-20 at 10 35 50@2x" src="https://github.com/Automattic/wp-calypso/assets/5634774/27750af3-01b2-48cf-9512-3f2e33388348">

Links on the left need to have a left aligned tooltip. But links on the right should be right aligned. Does that make sense?

Addison-Stavlo commented:
> The problem if we remove it is that the tooltip on the logo is no longer aligned properly:

Ok, thanks!  In that case I would suggest we add another 'position' label so that we can support both. Maybe `.tooltip-bottom`, `.tooltip-bottom-right`, `.tooltip-bottom-left` or something?

Addison-Stavlo committed:
keep left align tooltip styles for site item

Addison-Stavlo commented:
Updated and should be good for another test. Updated the description above:

> This updates this style definition in the global sidebar from `.tooltip-bottom` to `.tooltip-bottom-left` to signify left alignment, and is used with the WordPress.com menu item. Search and notif tooltips still default to `.tooltip-bottom`, but no longer get the left alignment overrides.

davemart-in reviewed (approved):
Looks good! Nice work.
